crate_universe/src/utils/starlark/select.rs

```diff
@@ -19,15 +24,24 @@ pub trait Select<T> {
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize, Clone)]
 pub struct SelectList<T: Ord> {
+    // Invariant: any T in `common` is not anywhere in `selects`.
     common: BTreeSet<T>,
+    // Invariant: none of the sets are empty.
     selects: BTreeMap<String, BTreeSet<T>>,
+    // Elements that used to be in `selects` before the most recent
+    // `remap_configurations` operation, but whose old configuration did not get
+    // mapped to any new configuration. They could be ignored, but are preserved
+    // here to generate comments that help the user understand what happened.
+    #[serde(skip_serializing_if = "BTreeSet::is_empty", default = "BTreeSet::new")]
+    unmapped: BTreeSet<T>,
```

Comment:
Was adding `unmapped` somehow required by this change, or is it an enhancement for documentation purposes? It looks like the latter (which is fine), but I want to make sure I understand if not.

Reply:
Keeping track of unmapped configurations isn't new, it was added in https://github.com/bazelbuild/rules_rust/pull/1647 (calling it to your attention because I don't see that you were involved in reviewing that PR).

The way that PR implemented it was:

```rust
impl<T: Ord + Clone> SelectList<T> {
    pub fn remap_configurations<'a, I, S>(
        &self,
        mapping: &'a BTreeMap<String, I>,
    ) -> (
        SelectList<WithOriginalConfigurations<T>>,
        BTreeMap<String, BTreeSet<T>>,
    )
```

where "unmapped" configurations are returned separately from SelectList, in the second element of the tuple.

We could continue to keep unmapped configurations out of `SelectList` and shuffle them around separately everywhere, like this:

```rust
pub struct RustLibrary {
    ...
    pub deps: (SelectList<WithOriginalConfigurations<String>>, BTreeMap<String, BTreeSet<String>>),
    pub proc_macro_deps: (SelectList<WithOriginalConfigurations<String>>, BTreeMap<String, BTreeSet<String>>),
    pub rustc_env: (SelectDict<WithOriginalConfigurations<String>>, BTreeMap<String, BTreeMap<String, T>>),
    pub rustc_flags: (SelectList<WithOriginalConfigurations<String>>, BTreeMap<String, BTreeSet<String>>),
```

It's just that... literally everywhere crate_universe wants to  serialize a SelectList, you also want to serialize all the configurations that got mapped away out of that select, to help the user understand what happened to them. So I've combined it together into one struct.